Applications/Spire/Source/Ui/ScrollBox.cpp

```diff
@@ -98,48 +111,74 @@ void ScrollBox::wheelEvent(QWheelEvent* event) {
   m_scrollable_layer->wheelEvent(event);
 }
 
-void ScrollBox::on_vertical_scroll(int position) {
-  m_body->move(m_body->pos().x(), -position);
-}
-
-void ScrollBox::on_horizontal_scroll(int position) {
-  m_body->move(-position, m_body->pos().y());
-}
-
 void ScrollBox::update_ranges() {
-  auto bar_width = [&] {
+  auto get_bar_width = [&] {
     if(m_scrollable_layer->get_vertical_scroll_bar().isVisible()) {
       return m_scrollable_layer->get_vertical_scroll_bar().width();
     }
     return 0;
-  }();
-  auto bar_height = [&] {
+  };
+  auto get_bar_height = [&] {
     if(m_scrollable_layer->get_horizontal_scroll_bar().isVisible()) {
       return m_scrollable_layer->get_horizontal_scroll_bar().height();
     }
     return 0;
-  }();
-  auto viewport_size = m_body->size() + QSize(bar_width, bar_height);
-  setMaximumSize(viewport_size);
+  };
+  auto border_size = get_border_size();
+  auto viewport_size = m_body->size() + border_size;
   if(m_vertical_display_policy == DisplayPolicy::ON_OVERFLOW) {
-    if(viewport_size.height() <= height()) {
+    if(viewport_size.height() <= height() - get_bar_height()) {
       m_scrollable_layer->get_vertical_scroll_bar().hide();
     } else {
       m_scrollable_layer->get_vertical_scroll_bar().show();
     }
   }
   if(m_horizontal_display_policy == DisplayPolicy::ON_OVERFLOW) {
-    if(viewport_size.width() <= width()) {
+    if(viewport_size.width() <= width() - get_bar_width()) {
       m_scrollable_layer->get_horizontal_scroll_bar().hide();
     } else {
       m_scrollable_layer->get_horizontal_scroll_bar().show();
     }
   }
-  auto vertical_range = std::max(m_body->height() - height() + bar_height, 0);
-  auto horizontal_range = std::max(m_body->width() - width() + bar_width, 0);
+  setMaximumSize(viewport_size + QSize(get_bar_width(), get_bar_height()));
```

Comment:
Because it exists in the original version, I just keep it here and add a border size in the calculation. I guess the intention is to prevent the ScrollBar from being bigger than its body. If the ScrollBar is bigger than its body, there are spaces between the border of the ScrollBox and its body.

Reply:
Okay, it's fine for now then.